src/ngraph_encapsulate_op.cc

```diff
@@ -229,9 +259,15 @@ class NGraphEncapsulateOp : public tf::OpKernel {
 
       // Create the nGraph output tensor
       void* dst_ptr = tf::DMAHelper::base(output_tensor);
-      auto t_result = m_ng_backend->create_tensor(elem_type, shape, dst_ptr);
+      if(m_ng_backend_name == "NNP") {
+      	auto t_result = m_ng_backend->create_tensor(elem_type, shape);
```

Comment:
For non-host backends, we shall only be calling `m_ng_backend->create_tensor(ng_element_type, ng_shape);` for the **first** time. Then we shall use the cached version of the created tensor. `m_ng_backend->create_tensor` is not free for non-CPU backends. We can afford to create tensors multiple times in CPU because the `src_ptr` is reused when we call `m_ng_backend->create_tensor(ng_element_type, ng_shape, src_ptr);`

We shall have a map of `<ng::Function, vector<ng::runtime::TensorView>> map_ng_function_to_inputs` to cache the input tensors.